arch/src/x86_64/smbios.rs

```diff
@@ -41,6 +44,7 @@ impl Display for Error {
             Clear => "Failure while zeroing out the memory for the SMBIOS table",
             WriteSmbiosEp => "Failure to write SMBIOS entrypoint structure",
             WriteData => "Failure to write additional data to memory",
+            ParseUuid(_) => "Failure to parse uuid, uuid format may be error",
```

Comment:
Since `uuid::Error` has `impl Error`. You can do `ParseUuid(e) => format!("Failure to parse uuid: {}",e),` and the underlying error will be shown to the user.